jujugui/static/gui/src/app/app.js

```diff
@@ -1300,7 +1301,7 @@ YUI.add('juju-gui', function(Y) {
           changes={currentChangeSet}
           charmsGetById={db.charms.getById.bind(db.charms)}
           deploy={utils.deploy.bind(utils, this)}
-          environment={db.environment}
+          env={env}
```

Comment:
This should be called model.
Also, we are passing a whole model connection to a component, just to be able to set the environmentName attribute. What do you think about 
```
setModelName={env.set.bind(env, 'environmentName')}
```
Or similar? So that the component can just call the given callable passing the model name.